File: EventPlus.Frontend/api/jwt/types/jwtResult.ts
import { CommandModel } from "~/api/command/types/commandModel";

export interface JwtResult {
  username: string;
  firstName: string;
  lastName: string | null;

  token: string;
  tokenExpires: Date;

  refreshToken: string;
  refreshTokenExpires: Date;

  avatar: string | null;
  commands: number[];
  lastActivityCommand: CommandModel | null;
}

File: EventPlus.Backend/EventPlus.Application/Services/Jwt/Models/JwtResult.cs
namespace EventPlus.Application.Services.Jwt.Models;

public sealed record JwtResult
{
    /// <example>aspadmin</example>
    public required string Username { get; init; }

    public required string FirstName { get; set; }
    public string? LastName { get; set; }

    /// <example>[JWT]</example>
    public required string Token { get; init; }

    public required string RefreshToken { get; init; }

    public DateTime TokenExpires { get; init; }
    public DateTime RefreshTokenExpires { get; init; }
    public string? Avatar { get; set; }

    public ICollection<long> Commands { get; set; }
    public long? LastActivityCommand { get; set; }
}